File: nodes/array-join/array-join.process.py
async def process(inputs, settings, config, nodeConfig):
  array = inputs.get("array", [])
  if not isinstance(array, list):
    array = [array]
  separator = str(inputs.get("separator", ","))
  
  return {
    "text": separator.join(str(x) for x in array)
  } 

File: sdks/nodejs/nodes/array-join/array-join.process.js
export default async ({inputs, settings, config}) => {

  const array = Array.isArray(inputs.array) ? inputs.array : [inputs.array];
  const separator = inputs.separator !== undefined ? String(inputs.separator) : ",";
  
  return {
    text: array.join(separator)
  };
}; 